File: Scripts/results/generated_code_results/Flutter/bak_code/Social Networking/SocialSoftware/FeedScreen.dart
import 'package:flutter/material.dart';
class FeedScreen extends StatefulWidget {
@override
_FeedScreenState createState() => _FeedScreenState();
}
class _FeedScreenState extends State<FeedScreen> {
final List<Map<String, dynamic>> _posts = [
{
'user': 'arochinski',
'content': 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',
'image': true,
'stats': {'answers': 7, 'likes': 59},
'timeAgo': '2 min'
},
{
'user': 'arochinski',
'content': 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',
'stats': {'answers': 7, 'likes': 59},
'timeAgo': '2 min'
},
{
'user': 'aura',
'content': 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',
'timeAgo': '1min'
},
];
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Image.asset('assets/images/template.png', height: 30),
actions: [
IconButton(
icon: Icon(Icons.menu),
onPressed: () {},
),
],
),
body: ListView.builder(
itemCount: _posts.length,
itemBuilder: (context, index) {
final post = _posts[index];
return Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
ListTile(
leading: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
title: Text(post['user']),
subtitle: Text(post['timeAgo']),
trailing: IconButton(
icon: Icon(Icons.more_horiz),
onPressed: () {},
),
onTap: () => Navigator.pushNamed(context, 'UserProfile'),
),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Text(post['content']),
),
if (post['image'] == true)
Padding(
padding: const EdgeInsets.all(16.0),
child: Image.asset('assets/images/template.png'),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Row(
children: [
IconButton(
icon: Icon(Icons.favorite_border),
onPressed: () {},
),
Text('${post['stats']['likes']}'),
],
),
Row(
children: [
IconButton(
icon: Icon(Icons.chat_bubble_outline),
onPressed: () {},
),
Text('${post['stats']['answers']}'),
],
),
IconButton(
icon: Icon(Icons.repeat),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.send),
onPressed: () {},
),
],
),
),
Divider(),
],
);
},
),
bottomNavigationBar: BottomNavigationBar(
type: BottomNavigationBarType.fixed,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),
BottomNavigationBarItem(icon: Icon(Icons.add_box), label: 'Create'),
BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Likes'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
onTap: (index) {
switch (index) {
case 1:
Navigator.pushNamed(context, 'Search');
break;
case 3:
Navigator.pushNamed(context, 'Activity');
break;
case 4:
Navigator.pushNamed(context, 'Profile');
break;
}
},
),
);
}
}

File: Scripts/results/generated_code_results/ablation_study/sam/Social Networking/SocialSoftware/FeedScreen.js
import React, { useState } from 'react';

import { View, Text, Image, StyleSheet, TouchableOpacity, FlatList, ScrollView } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import Icon from 'react-native-vector-icons/Ionicons';



const Post = ({ post, onUserPress }) => (

  <View style={styles.post}>

    <TouchableOpacity onPress={() => onUserPress(post.user)}>

      <Text style={styles.username}>{post.user}</Text>

    </TouchableOpacity>

    <Text style={styles.content}>{post.content}</Text>

    {post.image && (

      <Image source={require('../assets/snack-icon.png')} style={styles.postImage} />

    )}

    <View style={styles.postStats}>

      <View style={styles.postStats}>
        <Text>{post.stats?.answers || 0} answers</Text>
        <Text>{post.stats?.likes || 0} likes</Text>
        <Text>{post.timeAgo}</Text>
      </View>

    </View>

  </View>

);



const FeedScreen = () => {

  const navigation = useNavigation();

  const [posts, setPosts] = useState([

    {

      user: 'arochinski',

      content: 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

      image: true,

      stats: { answers: 7, likes: 59 },

      timeAgo: '2 min'

    },

    {

      user: 'arochinski',

      content: 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

      stats: { answers: 7, likes: 59 },

      timeAgo: '2 min'

    },

    {

      user: 'aura',

      content: 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

      timeAgo: '1min'

    }

  ]);



  const handleUserPress = (username) => {

    navigation.navigate('UserProfile', { username });

  };



  const handleNavigation = (screen) => {

    navigation.navigate(screen);

  };



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <Image source={require('../assets/snack-icon.png')} style={styles.logo} />

        <TouchableOpacity onPress={() => handleNavigation('Activity')}>

          <Icon name="notifications-outline" size={24} color="#000" />

        </TouchableOpacity>

      </View>

      <FlatList

        data={posts}

        renderItem={({ item }) => <Post post={item} onUserPress={handleUserPress} />}

        keyExtractor={(item, index) => index.toString()}

      />

      <View style={styles.bottomNav}>

        <TouchableOpacity onPress={() => handleNavigation('Feed')}>

          <Icon name="home" size={24} color="#000" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => handleNavigation('Search')}>

          <Icon name="search" size={24} color="#000" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => {}}>

          <Icon name="add-circle-outline" size={24} color="#000" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => {}}>

          <Icon name="heart-outline" size={24} color="#000" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => handleNavigation('Profile')}>

          <Icon name="person" size={24} color="#000" />

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#fff',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    padding: 10,

    borderBottomWidth: 1,

    borderBottomColor: '#e0e0e0',

  },

  logo: {

    width: 30,

    height: 30,

  },

  post: {

    padding: 10,

    borderBottomWidth: 1,

    borderBottomColor: '#e0e0e0',

  },

  username: {

    fontWeight: 'bold',

    marginBottom: 5,

  },

  content: {

    marginBottom: 10,

  },

  postImage: {

    width: '100%',

    height: 200,

    resizeMode: 'cover',

    marginBottom: 10,

  },

  postStats: {

    flexDirection: 'row',

    justifyContent: 'space-between',

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    borderTopWidth: 1,

    borderTopColor: '#e0e0e0',

    paddingVertical: 10,

  },

});



export default FeedScreen;